libmamba/include/mamba/core/solver.hpp

```diff
@@ -60,6 +60,16 @@ namespace mamba
     {
     public:
 
+        struct Flags
```

Comment:
For future maintainance and optimization, it might be best to make this expose only functions to read or write the values, as the implementation could change (to bits for example, in particular if we add more values later). Changing the implementation would then be easy without changing the interface.
An alternative, IFF flags are always boolean, would be to use an enum as key and have 1 functoin for reading, 1 functoin for storing.

Reply:
> An alternative, IFF flags are always boolean, would be to use an enum as key and have 1 functoin for reading, 1 functoin for storing.

That would be nice, but I don't want to limit the future of the flags to boolean only.

> For future maintainance and optimization, it might be best to make this expose only functions to read or write the values, as the implementation could change (to bits for example, in particular if we add more values later). Changing the implementation would then be easy without changing the interface.

I would agree in general, but in this case I think there is value in just having a plain, dumb, data only class.
I don't the complexity of storing flags as bits  would ever make sense for such high-level flags.